cloud/services/agentpools/client.go

```diff
@@ -51,7 +51,7 @@ func NewClient(auth azure.Authorizer) *AzureClient {
 func newAgentPoolsClient(subscriptionID string, baseURI string, authorizer autorest.Authorizer) containerservice.AgentPoolsClient {
 	agentPoolsClient := containerservice.NewAgentPoolsClientWithBaseURI(baseURI, subscriptionID)
 	agentPoolsClient.Authorizer = authorizer
-	agentPoolsClient.AddToUserAgent(azure.UserAgent())
+	_ = agentPoolsClient.AddToUserAgent(azure.UserAgent()) // intentionally ignore error as it doesn't matter
```

Comment:
Would you consider refactoring client initialization into a func rather than repeating the linter fix all over? Perhaps, like the following.

```go
// newDisksClient creates a new disks client from subscription ID.
func newDisksClient(subscriptionID string, baseURI string, authorizer autorest.Authorizer) compute.DisksClient {
	disksClient := compute.NewDisksClientWithBaseURI(baseURI, subscriptionID)
	azure.SetAutoRestClientDefaults(&disksClient.Client, authorizer)
	return disksClient
}

// in the `azure` package; perhaps, ./cloud/defaults.go with a test in ./cloud/defaults_test.go
func SetAutoRestClientDefaults(c *autorest.Client, auth autorest.Authorizer) {
	c.Authorizer = auth
	_ = c.AddToUserAgent(azure.UserAgent()) // intentionally ignore error as it doesn't matter
}

```

Reply:
I get it. 👍